src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.service.ts

```diff
@@ -6,47 +6,45 @@ import { DotLayoutColumn } from '../../shared/models/dot-layout-column.model';
 import * as _ from 'lodash';
 import { DotPageView } from '../../shared/models/dot-page-view.model';
 import { NgGridItemConfig } from 'angular2-grid';
+import { DOT_LAYOUT_GRID_NEW_ROW_TEMPLATE } from '../../shared/models/dot-layout.const';
 
 /**
- * Provide methods to transform data types, to be able to display it in the NgGrid.
+ * Provide methods to transform NgGrid model into PageView model and viceversa.
  *
- * @class DotEditLayoutGridService
+ * @class DotEditLayoutService
  */
 @Injectable()
-export class DotEditLayoutGridService {
+export class DotEditLayoutService {
     constructor() {}
 
     /**
-     * Transform and Object from  DotPageView to DotLayoutGridBox to be able to display it on the Grid
+     * Transform and return DotPageView to DotLayoutGridBox
      *
      * @param {DotPageView} dotPageView
      * @param {NgGridItemConfig} newRow
      * @returns {DotLayoutGridBox[]}
      */
-    getDotLayoutGridBox(dotPageView: DotPageView, newRow: NgGridItemConfig): DotLayoutGridBox[] {
+    getDotLayoutGridBox(dotPageView: DotPageView): DotLayoutGridBox[] {
         const grid: DotLayoutGridBox[] = [];
+
         dotPageView.layout.body.rows.forEach((row, rowIndex) => {
             row.columns.forEach(column => {
                 grid.push({
-                    containers: column.containers.map(
-                        containerId =>
-                            dotPageView.containers[containerId]
-                                ? dotPageView.containers[containerId].container
-                                : containerId
-                    ),
-                    config: Object.assign({}, newRow, {
+                    containers: column.containers.map(containerId => dotPageView.containers[containerId].container),
+                    config: Object.assign({}, DOT_LAYOUT_GRID_NEW_ROW_TEMPLATE, {
```

Comment:
I make ```newRow``` "global" and that way we don't need to pass it as a param, besides it's something that this method needs all the time.